### PR TITLE
RD-4403: Creating secrets on terraform modal submission - hiding secr…

### PR DESCRIPTION
**Pull request by jawamartus**
…ets with asterisks

## Description

RD-4403

Covering missing requirements.

## Screenshots / Videos

image

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

[Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/2760/)

## Documentation

N/A

### CODE REVIEW

File: widgets/common/src/terraformModal/TerraformVariableNameInput.tsx [hunk 1]
```diff
@@ -6,17 +6,17 @@ import { inputMaxLength } from './TerraformModal';
 
 const t = Stage.Utils.getT('widgets.blueprints.terraformModal.variablesTable');
 
-interface TerraformVariableValueInputProps extends CustomConfigurationComponentProps<string> {
+interface TerraformVariableNameInputProps extends CustomConfigurationComponentProps<string> {
     rowValues?: Variable;
 }
 
-export default function TerraformVariableValueInput({
+export default function TerraformVariableNameInput({
```
Comment (jawamartus) [inline widgets/common/src/terraformModal/TerraformVariableNameInput.tsx:13]: Refactoring

Comment (qooban) [reply widgets/common/src/terraformModal/TerraformVariableNameInput.tsx:13]: Does this component show Terraform variable name input? It didn't show Terraform variable value input, so good that you spotted it, but it seems to me that it shows an input field to provide either secret name or input name. These are Cloudify resources' names, so maybe more accurate would be `CloudifyResourceNameInput`?

File: widgets/common/src/terraformModal/TerraformModal.tsx [hunk 2]
```diff
@@ -61,6 +61,24 @@ function LengthLimitedDynamicTableInput({ name, onChange, ...rest }: CustomConfi
     );
 }
 
+interface TerraformVariableValueInputProps extends CustomConfigurationComponentProps<string> {
+    rowValues?: Variable;
+}
+
+function TerraformVariableValueInput({ name, onChange, rowValues, ...rest }: TerraformVariableValueInputProps) {
```
Comment (qooban) [inline widgets/common/src/terraformModal/TerraformModal.tsx:68]: Seeing that `LengthLimitedDynamicTableInput` and `TerraformVariableValueInput` components are quite similar (the only difference is in `Input.type`) did you consider having one common component for it?

File: widgets/common/src/terraformModal/TerraformVariableNameInput.tsx (hunk 1 above)
Comment (jawamartus) [reply widgets/common/src/terraformModal/TerraformVariableNameInput.tsx:13]: The column is described as name in the design. The variables and environmental variables have:
`key`
`source`
`name`
`value`

File: widgets/common/src/terraformModal/TerraformModal.tsx (hunk 2 above)
Comment (jawamartus) [reply widgets/common/src/terraformModal/TerraformModal.tsx:68]: The components are not presentational components but more like  logic components (google for the concepts if you are not familiar with it). I cannot see duplication of knowledge here. 
What is more LengthLimitedDynamicTableInput shouldn't be aware about rowValues object.

There is plenty things to improve in this file. It's definitely to big. I decided to keep it as it is for know. At least until we don't have specific agreement where we should place side-effects and business logic and how those should be separated from UI components. 

Also I don't think that making one common component for those fields is a good practice.

Comment (jawamartus) [reply widgets/common/src/terraformModal/TerraformModal.tsx:68]: Answering your question. Yes, I did consider this.

**Comment by jawamartus** [comment]
The system tests fails are not related to the changes in this PR.

File: widgets/common/src/terraformModal/TerraformVariableNameInput.tsx (hunk 1 above)
Comment (qooban) [reply widgets/common/src/terraformModal/TerraformVariableNameInput.tsx:13]: IMHO:
1. the component name is kind of a shortcut for component's content, so it should as much as possible describe it
(what will developer think when he's reading `TerraformVariableNameInput`? A component providing an input field for Terraform variable name)
2. the context in which a component is used is less relevant and should not decide what is the component name (what if we would like to use the same component in two different contexts?)

Also, in the list you have presented (`key`, `source`, `name`, `value`), Terraform variable name is actually a `key`.

File: widgets/common/src/terraformModal/TerraformModal.tsx (hunk 2 above)
Comment (qooban) [reply widgets/common/src/terraformModal/TerraformModal.tsx:68]: OK.

**Review by qooban — APPROVED** [review]

**Pull request merged**